### PR TITLE
chore: deprecated debian stretch specific files and scripts are removed

### LINKED ISSUE #13969: Assorted packer and dev-vm cleanups
- [x] Remove old libvirt images and clean up names
- [x] Grep for stretch only stuff and remove it in vagrant and ansible
- [x] Investigate the error message that sometimes appears but doesn't seem to have an effect. If you cannot reproduce it any more after destroying and recreating the machine, ignore this task.
```
==> magma: Machine booted and ready!
==> magma: Checking for guest additions in VM...
==> magma: Setting hostname...
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

systemctl stop ifup@ovs-system.service;systemctl start ifup@ovs-system.service

Stdout from the command:



Stderr from the command:

A dependency job for ifup@ovs-system.service failed. See 'journalctl -xe' for details.
```

### PR DESCRIPTION
**Pull request by mpfirrmann**
## Summary

The AGW VMs now run on Ubuntu 20.04 and as the last part of the migration from Debian to Ubuntu, we purge the repo for leftover files, scripts, and comments. As part of the cleanup we ...

- removed the deprecated packer templates with libvirt as VM provider
- removed the deprecated packer templates based on Debian "stretch"
- removed the deprecated AWS images based on Debian "stretch"
- removed the deprecated Ansible role `stretch_snapshot`
- removed the deprecated `provision_stretch.sh` script
- removed several hints and mentions of the deprecated machines
- removed the deprecated packer templates using Ubuntu 16.04 "xenial"
- removed the temporary packer template that mitigated a broken OVS
- removed an incomplete packer template
- renamed the most recent `magma-dev` VM template: `magma-focal-virtualbox.json` -> `magma-dev-virtualbox.json`
- removed `setup.sh` because all steps are included in the now used `ubuntu_setup.sh`
- removed `ansible_debian.sh` due to its references to Ubuntu 14.04 "trusty"
- removed the deprecated `.seed` file for the debian packer templates
- removed several mentions of debian OS in Ansible roles
- removed the unused `envoy` Ansible role referencing the Debian "stretch" repo of "Getenvoy.io"
- removed the deprecated `uselocalpkgrepo` Ansible role
- readability of `magma-dev-virtualbox.json` is improved
- hints to "old" `magma-focal` are adapted

Closes #13969.
## Test Plan

Smoke testing before merging:
Pack `magma-dev` VM locally
Full provision via Vagrant
-> see also [issue/13566](https://github.com/magma/magma/issues/13566)
- [x] Python unit tests
- [x] Build AGW
- [x] Run `test_attach_detach` and `test_attach_dl_tcp_data`

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Comment by mpfirrmann** [comment]
Grepping for `stretch` also gives me mentions in several other directories / files:

`.cache/go/pkg/mod/github.com/docker/docker@v1.13.1/ ...`
`.cache/go/pkg/mod/google.golang.org/genproto@v.0.0.0-202108223120-3a666f561d7aa/ ...`
`ci-scripts/registry/aptly/deploy_image.sh`
`experimental/cloudstrapper/playbooks/roles/magma-agw/vars/main.yaml`
`hil_testing/Magma_Ansible/project/roles/upgrade-prep/tasks/main.yaml`
`nms/app/components/JsonEditor.tsx`
`lte/gateway/release/upgrade_magma.sh`
`lte/gateway/release/pydep`
`lte/gateway/release/deb_dependencies.bzl`

I'm not quite sure what can be touched without breaking anything.

**Review by sebathomas — APPROVED** [review]

**Review by sebathomas — APPROVED** [review]

**Review by crasu — APPROVED** [review]

File: orc8r/tools/packer/magma-test-virtualbox.json [hunk 1]
```diff
@@ -89,11 +89,6 @@
       "script": "scripts/vagrant_key.sh",
       "type": "shell"
     },
-    {
-      "execute_command": "echo 'vagrant' | {{.Vars}} sudo -S -E bash '{{.Path}}'",
-      "script": "scripts/setup.sh",
```
Comment (nstng) [inline orc8r/tools/packer/magma-test-virtualbox.json:94]: This was removed with the comment that `ubuntu_setup.sh` already covers all the steps - I do not see `ubuntu_setup.sh` called here directly. Is it called implicitly?

**Comment by nstng** [comment]
overall LGTM, thank you for the needed clean-up!
`lte/gateway/deploy/roles/uselocalpkgrepo` also looks stretch related, but is still used in `orc8r/tools/packer/magma-test-virtualbox.json`. can this be removed too?

File: orc8r/tools/packer/magma-test-virtualbox.json (hunk 1 above)
Comment (mpfirrmann) [reply orc8r/tools/packer/magma-test-virtualbox.json:94]: It's called in L71.

Comment (nstng) [reply orc8r/tools/packer/magma-test-virtualbox.json:94]: woops, (what did I search for?), merci

**Review by nstng — APPROVED** [review]

**Review by MoritzThomasHuebner — COMMENTED** [review]
LGTM

File: orc8r/tools/packer/magma-dev-virtualbox.json [hunk 2]
```diff
@@ -1,115 +1,127 @@
...
-      "pause_before": "10s"
-    }, {
-      "type": "shell",
-      "script": "scripts/vagrant_key.sh",
-      "execute_command": "echo 'vagrant' | sudo -S env {{.Vars}} {{.Path}}"
+      "scripts": [
+        "scripts/ubuntu_setup.sh",
+        "scripts/vagrant_key.sh"
+      ],
+      "type": "shell"
```
Comment (jheidbrink) [inline orc8r/tools/packer/magma-dev-virtualbox.json]: nitpick: I find the old ordering that was based on semantics easier to understand than the new alphabetical ordering.

For example a provisioning step is easier to read for me if `type` comes first, followed by the most important argument to that type, like `inline` or `script`. Same for other items in the JSON hierarchy like the builders where I would also prefer having the type at the top, or having the post-processors after the provisioners which sort of reflects the order in which things are done.

Sticking to the old ordering has the additional advantage of making this diff much nicer to read ;)

Comment (mpfirrmann) [reply orc8r/tools/packer/magma-dev-virtualbox.json]: Adapted. While we only renamed the `.json` here, I agree that a semantic ordering improves readability and leads to an easier understanding.

**Review by jheidbrink — APPROVED** [review]

File: orc8r/tools/packer/magma-dev-virtualbox.json (hunk 2 above)
Comment (jheidbrink) [reply orc8r/tools/packer/magma-dev-virtualbox.json]: Thanks! And yes, I actually read the diff wrong because of the rename ;)

**Pull request merged**